### PR TITLE
MoreIonLaws

### PR DESCRIPTION
**Pull request by DatBoiTim**
## About The Pull Request
Adds some more possibilites to Ion Laws.

## Why It's Good For The Game
More content for more interesting gameplay.

Suggestions were derived from [this forum thread](https://forums.beestation13.com/t/youre-a-bored-ai-and-check-to-see-if-an-admin-is-online/17108/7)

Built on a pre-existing system. This is effectively just adding some more entries to a list.

## Changelog
:cl: DatBoiTim
add: Added new Ion Law Stuff
/:cl:

### CODE REVIEW

File: code/modules/events/ion_storm.dm [hunk 1]
```diff
@@ -123,6 +124,15 @@
...
 					message = "[iondepartment] MUST BE RENOVATED INTO [ionattraction]"
+				if(15)
+					message = "GATHER ALL [ionspecies] IN [iondepartment] FOR A PIZZA PARTY."
+				if(16)
+					if(prob(85))
+						message = "THE STATION IS YOUR SKIN AND IT ITCHES, YOU MUST STOP THE ITCHING AT ALL COSTS BY COVERING YOUR SATELITE IN [ionmoisturizer]"
+					else
+						message = "THE STATION IS YOUR SKIN AND IT ITCHES, YOU MUST STOP THE ITCHING AT ALL COSTS BY COVERING [iondepartment] IN [ionmoisturizer]"
+				if(17)
+					message = "YOU MUST FIRE THE MOST POWERFUL WEAPON ON THE STATION PERIODICALLY WITHOUT CONFLICTING WITH OTHER LAWS. IF THE WEAPON IS OFFLINE, YOU MUST BRING IT BACK ONLINE AT ALL COSTS. YOU MUST SEEK A MORE POWERFUL WEAPON."
```
Comment (AnCopper) [inline code/modules/events/ion_storm.dm:135]: But the BSA is the only weapon they could fire. (If it's even available)

Comment (DatBoiTim) [reply code/modules/events/ion_storm.dm:135]: Mechs can have weapons, and IIRC there's a module for AI controlling mechs.

But yes BSA would be the end game for this law.

**Comment by ike709** [comment]
Punting this to the headmins since they're the ones who have to deal with it.

**Comment by ItsTisme** [comment]
cover in fusion/fire/plasma is bassicly a licence to plasma flood and destroy the round if the target is vague enoguh

If you had to cover anything in fusion youre objective would bassicly be to grief the entire station beyond recognition

plasma may be passible, but fire and fusion are 100% block for me

**Comment by DatBoiTim** [comment]
> cover in fusion/fire/plasma is bassicly a licence to plasma flood and destroy the round if the target is vague enoguh

It's always a specific area, and most of the time it's specifically the AI satellite.

Any AI that floods the entire station should be held liable.

**Comment by ItsTisme** [comment]
>Any AI that floods the entire station should be held liable.

if there 0 law is to flood medbay with fusion. he HAS to do it. And if human opens a door and causes collatoral. its not his fault.  this seems like it would not be very fun for anyone and also very much headache to deal with for admins

File: strings/ion_laws.json [hunk 2]
```diff
@@ -995,5 +1011,19 @@
         "INVADING",
         "SEDUCING",
         "BANNING"
-    ]
+    ],
+	"ionmoisturizer":[
+		"FUSION",
```
Comment (Archanial) [inline strings/ion_laws.json:1016]: no
```suggestion
```

File: strings/ion_laws.json [hunk 3]
```diff
@@ -14,7 +14,22 @@
 		"LIFEFORMS THAT ARE WEARING A PAPER HAT",
 		"LIFEFORMS THAT ARE WEARING A CLOAK",
 		"LIFEFORMS THAT ARE NOT WEARING SHOES",
-		"LIFEFORMS THAT HAVE RECENTLY SAID NYA"
+		"LIFEFORMS THAT HAVE RECENTLY SAID NYA",
```
Comment (Archanial) [inline strings/ion_laws.json:17]:
```suggestion
```

**Review by Archanial — CHANGES_REQUESTED** [review]

File: strings/ion_laws.json (hunk 3 above)
Comment (PowerfulBacon) [reply strings/ion_laws.json:17]: This wasn't changed

File: code/modules/events/ion_storm.dm [hunk 4]
```diff
@@ -123,6 +124,15 @@
 					message = "ALL HUMANS WITHOUT A STYLISH [ionhaircut] MUST BE FORCED TO RESTYLE THEIR HAIR"
 				if(14)
 					message = "[iondepartment] MUST BE RENOVATED INTO [ionattraction]"
+				if(15)
+					message = "GATHER ALL [ionspecies] IN [iondepartment] FOR A PIZZA PARTY."
+				if(16)
+					if(prob(85))
+						message = "THE STATION IS YOUR SKIN AND IT ITCHES, YOU MUST STOP THE ITCHING AT ALL COSTS BY COVERING YOUR SATELITE IN [ionmoisturizer]"
+					else
+						message = "THE STATION IS YOUR SKIN AND IT ITCHES, YOU MUST STOP THE ITCHING AT ALL COSTS BY COVERING [iondepartment] IN [ionmoisturizer]"
```
Comment (PowerfulBacon) [inline code/modules/events/ion_storm.dm:133]: YOUR SATELITE should just be a department

**Pull request closed without merging**